PIP-249 site redesign

visortelle opened the pull request:
### Documentation

<!-- DO NOT REMOVE THIS SECTION. CHECK THE PROPER BOX ONLY. -->

- [ ] `doc` <!-- Your PR contains doc changes. Please attach the local preview screenshots (run `./preview.sh` at root path) to your PR description, or else your PR might not get merged. -->
- [ ] `doc-required` <!-- Your PR changes impact docs and you will update later -->
- [x] `doc-not-needed` <!-- Your PR changes do not impact docs -->
- [ ] `doc-complete` <!-- Docs have been already added -->

PIP-249: https://github.com/apache/pulsar/issues/19611

Live deployment: https://pulsar-site-pip-249.vercel.app/

Visible changes:
- The Home page.
- Docs page.
- Blog page.
- Ecosystem page.
- Case studies page.
- Resources page.
- Events page.

What’s left:
- [x] Ensure that homepage animation performance is ok, otherwise, replace it with something else. 
- [x] Also, we need to decide what to do with the animation in Firefox. See attached screenshot. Maybe use a static illustration variant for Firefox?
- [x] Announcement bar. The styles are ready. Need to integrate it with docusaurus.config.js (simple). ~0.5h.
- [x] Powered by page. ~0.5h.
- [x] Community page. ~3h.
I’ll resolve these points while @tisonkun reviews the changes.
- [x] Fill users' testimonials with actual content. https://github.com/tealtools/pulsar-site-pip-249/blob/02c9096a0a3da557b7b779bb48bb4aa8f4066ff3/data/testimonials.ts#L7

Needs recheck:
- [x] Contribute page and the recent Client Feature Matrix page.
- [ ] Maybe some hidden pages I don’t know about. I checked all listed under src/pages/*.

asafm commented:
![image](https://github.com/apache/pulsar-site/assets/989425/cb55d867-47e7-4896-a933-69293c5dff58)
This is still at work, right?

visortelle commented:
@asafm please check now.

I think there is a mistake in user testimonials. K.Merkushev is duplicated and J.Joseph is a company.
Maybe you know the right info?

<img width="655" alt="Screenshot 2023-05-14 at 4 47 24 PM" src="https://github.com/apache/pulsar-site/assets/9302460/7f6ecfd1-db0f-4c69-ad1b-4d1c04de7cbb">

https://github.com/apache/pulsar-site/blob/70dc8e54ccf16308875c4345b28924e79700ce5b/src/pages/index.js#L101

visortelle commented:
Didn't found this info on LinkedIn, but Google helped.

<img width="442" alt="Screenshot 2023-05-14 at 4 50 52 PM" src="https://github.com/apache/pulsar-site/assets/9302460/1474815a-bbad-4780-9246-5bfb9a6e2286">

visortelle commented:
I noticed issues on mobile. Something goes wrong with background positions when building for production. Will try to figure out the reason.

Local dev:

<img width="767" alt="Screenshot 2023-05-14 at 4 56 18 PM" src="https://github.com/apache/pulsar-site/assets/9302460/84157433-ccb3-4213-99ad-c56ba3323248">

Production build:

<img width="727" alt="Screenshot 2023-05-14 at 4 56 33 PM" src="https://github.com/apache/pulsar-site/assets/9302460/9baa1ece-b5c6-43c7-be03-5170e7fba49f">

visortelle commented:
Fixed my previous comment issue.

asafm commented:
![image](https://github.com/apache/pulsar-site/assets/989425/dce0a6d6-45bc-4ab8-89b9-9612bbbd540e)
The font on the left navigation bar is the same as the one on the main text?

asafm commented:
![image](https://github.com/apache/pulsar-site/assets/989425/49b6ba63-8870-4a23-b2f2-199e066ec3c6)
This section is still work in progress?

asafm commented:
When surfing on iPhone, the icons for Pulsar features sometime is cut:
![image](https://github.com/apache/pulsar-site/assets/989425/d3dad902-43c3-4775-b3a3-72f999974ef4)

visortelle commented:
@tisonkun can you please provide instructions on how to develop abd build ‘/contribute’ and other pages that aren’t available automatically after running ‘npm start’

asafm commented:
In community, something is not aligned on iPhone 13 mini: 
![image](https://github.com/apache/pulsar-site/assets/989425/b3516ff6-2094-4860-9507-17851753b326)

asafm commented:
The comment up is actually for the entire community page

asafm commented:
Other than that, it's perfect!

visortelle commented:
> The font on the left navigation bar is the same as the one on the main text?

Yes, the font-family is the same.

Made it less bold. Probably now it looks better.

<img width="1440" alt="Screenshot 2023-05-14 at 6 10 11 PM" src="https://github.com/apache/pulsar-site/assets/9302460/3e298d2b-e381-476a-9504-c772916a7710">

visortelle commented:
@tisonkun maybe there is any way to build a full site in Docker?

I want to be able to build a site in a single command and don't think about tools, or their versions.

Or maybe you have such plans? If yes, I'd be happy to help.

tisonkun commented:
The file you link is quite old and I don't think it helps here.

https://github.com/apache/pulsar/issues/20272 tracks your need IIUC and let me prepare a patch today against this repo and ping you.

tisonkun commented on docusaurus.config.js:
```diff
@@ -124,298 +125,321 @@ const injectLinkParseForEndpoint = ([, info]) => {
+  ],
+  plugins: [
+    [
+      "content-docs",
```
It seems we are missing the `contribute` instance here:

```js
        [
            'content-docs',
            /** @type {import('@docusaurus/plugin-content-docs').Options} */
            ({
                id: 'contribute',
                path: 'contribute',
                routeBasePath: 'contribute',
                showLastUpdateAuthor: true,
                showLastUpdateTime: true,
                sidebarPath: require.resolve('./sidebarsDevelopment.js'),
                editUrl: `${githubSiteUrl}/edit/main`,
            }),
        ],
```

visortelle replied:
Yes, thank you!
🤦‍♂️

visortelle commented:
> This section is still work in progress?

@asafm Fixed.

https://pulsar-site-pip-249.vercel.app/contribute/

visortelle commented:
Will chech 

> Build failed in CI. Perhaps you need to run yarn install and update the lockfile?

Will check in an hour.

tisonkun commented on versioned_docs/version-2.9.x/about.md:
```
Compiled with problems:
×
ERROR in ./versioned_docs/version-3.0.x/about.md 1:1096-1152
Module not found: Error: Can't resolve '@site/src/components/BlockLinks' in '/Users/chenzili/Brittani/pulsar-site/versioned_docs/version-3.0.x'
ERROR in ./versioned_docs/version-3.0.x/about.md 1:1152-1206
Module not found: Error: Can't resolve '@site/src/components/BlockLink' in '/Users/chenzili/Brittani/pulsar-site/versioned_docs/version-3.0.x'
```

Need for 3.0 also I think.

tisonkun commented on .github/workflows/ci-precommit.yml:
```diff
@@ -17,8 +17,8 @@
 on:
-  pull_request:
-    branches: [main]
+  push:
```
I suggest you revert these commits. You can open a pull request against `main` in your fork to trigger CI also.

tisonkun replied:
Otherwise we cannot run CI upstream.

visortelle replied:
Reverted :)

tisonkun commented:
Maybe you can tick `Allow edits by maintainers` box so that I can try to fix the build issue for you.

visortelle commented:
2h+. Wow.

<img width="1033" alt="Screenshot 2023-05-15 at 6 45 28 PM" src="https://github.com/apache/pulsar-site/assets/9302460/5d43e363-8d91-4b36-893f-02fe06ba3181">

@tisonkun I assume, the next builds should take much less time?

There are a few minor tweaks by the designer.
I assume it would be preferable to make new changes in new PRs. 😄

tisonkun commented:
> I assume, the next builds should take much less time?

This is another tech debt that we maintain too much versions and to build them at once exhaust GitHub Actions Runner memory so we use [`split-version-build.sh`](https://github.com/apache/pulsar-site/blob/22081daf8095676f897a0224351e6116d7d0d183/scripts/split-version-build.sh) to build changed versions one by one. If only a few versions touched, it can be much faster.

Ideally, legacy versions should be archived as a snapshot, which can be another improvement proposal to work on.

tisonkun commented:
LGTM. I think we can merge this patch to prevent further conflicts. Following improvement can be done iteratively.

@asafm @visortelle what do you think? Is there any blocker from merging this as a stable base version?

One thing I can refer to is that @visortelle perhaps you can show this patch and the demo site on dev@pulsar.apache.org so that it doesn't surprise developers that the look of the website is changed.

visortelle commented:
@tisonkun great 👍

I'll prepare the announcement.

Anonymitaet commented:
@visortelle thanks for your great contribution. The new look is more clean and professional! 

I have some questions about the doc layout. Can you take a look? Thank you!

### Font and line spacing

The texts look a little dense. What are the default text size and line spacing? Does it make sense to increase them a bit to increase reliability?

https://pulsar-site-pip-249.vercel.app/docs/next/develop-tools/

<img width="984" alt="image" src="https://github.com/apache/pulsar-site/assets/50226895/5c6064b5-ef97-400b-b307-2691b2957534">

### Font

Do we need to keep consistent on "regular" or "bold"?

https://pulsar-site-pip-249.vercel.app/docs/next/concepts-replication/#replication-patterns

<img width="1745" alt="image" src="https://github.com/apache/pulsar-site/assets/50226895/143fa3fc-7fb5-46f1-9fb2-4007da2457fb">

### Color

https://pulsar-site-pip-249.vercel.app/docs/next/concepts-throttling/

<img width="899" alt="image" src="https://github.com/apache/pulsar-site/assets/50226895/47a2da3d-954b-471e-8c97-45ee7895cbd2">

Anonymitaet commented:
As suggested by @codelipenghui and @mattisonchao, it's more reasonable to update this value to 10ms.

https://pulsar-site-pip-249.vercel.app/
<img width="1068" alt="image" src="https://github.com/apache/pulsar-site/assets/50226895/2debc854-c95e-427c-a3c7-0cda3822c36b">

Data source:
<img width="487" alt="image" src="https://github.com/apache/pulsar-site/assets/50226895/d099ad22-15ce-455b-b825-0bc1751b8f6b">

Anonymitaet commented:
As suggested by @codelipenghui @mattisonchao @BewareMyPower, it's more reasonable to update this value to 10ms.

https://pulsar-site-pip-249.vercel.app/
<img width="1027" alt="image" src="https://github.com/apache/pulsar-site/assets/50226895/da17a5bc-83b7-4b4f-a229-c5ba3b1c4d71">

visortelle commented:
@Anonymitaet thank you for the review. 😀

> The texts look a little dense. What are the default text size and line spacing? Does it make sense to increase them a bit to increase reliability?

The text size is 16px. I increased the line-height a bit from 24px to 26px (1.625).

> Do we need to keep consistent on "regular" or "bold"?

h1 and h2 headers have regular fort by design.

<img width="687" alt="Screenshot 2023-05-19 at 7 36 01 AM" src="https://github.com/apache/pulsar-site/assets/9302460/30bc9c3c-f374-4db8-b250-de67cecb7be7">

https://www.figma.com/file/0jERoA2DTlfQoj1uH09FxQ/Pulsar-website?type=design&node-id=254-422&t=0DUnBCdzrLQVUWdj-0

For me, it looks normal.
Probably we can keep it for other pages, and make the bold for docs pages. 
@emidio-cardeira WDYT?

> Does it make sense to change "note" background color to more prominent one (e.g. blue) since it requests attention?

Docusaurus has a special syntax for that: https://docusaurus.io/docs/markdown-features/admonitions.

<img width="812" alt="Screenshot 2023-05-19 at 7 54 17 AM" src="https://github.com/apache/pulsar-site/assets/9302460/4607f61e-04c4-4027-8fa1-d58ab07ff717">

Using `:::info`

<img width="848" alt="Screenshot 2023-05-19 at 7 51 40 AM" src="https://github.com/apache/pulsar-site/assets/9302460/32689d63-99f1-4aa5-80a6-8329150ac1d2">

Using `:::caution`

<img width="841" alt="Screenshot 2023-05-19 at 7 52 36 AM" src="https://github.com/apache/pulsar-site/assets/9302460/bb393f8b-032d-43ac-8953-8772122eac2b">

For me, it makes sense to distinguish blocks visually that provide more details or give pro tips, and blocks that contain information about potentially destructive actions.
@asafm WDYT?

I found 244 `:::note` entries in the `/docs` dir. Not too many. 
@Anonymitaet if you want, I can help you with changing it to tip/info/caution/danger.

> Tables. Update this light gray to a little darker one?

I updated it to the gray color that is used in other places. For me, contrast is enough. :)

<img width="833" alt="Screenshot 2023-05-19 at 8 05 59 AM" src="https://github.com/apache/pulsar-site/assets/9302460/3139baf1-14af-4513-8c87-e7e6ecc25e32">

Higher contrast doesn't look good.

<img width="840" alt="Screenshot 2023-05-19 at 8 13 35 AM" src="https://github.com/apache/pulsar-site/assets/9302460/15118ab1-92e3-4a83-b1d8-31d2ac6dedaf">

@emidio-cardeira WDYT?

> Logo. Why change this from blue to black?

Maybe @emidio-cardeira or @asafm can give a good answer here.

>  it's more reasonable to update this value to 10ms.
Updated.

visortelle commented:
@Anonymitaet I replaced all `:::note` to `:::info`. Now it's highlighted with blue color.

<img width="1440" alt="Screenshot 2023-05-19 at 10 01 17 AM" src="https://github.com/apache/pulsar-site/assets/9302460/89a15198-f661-4c9d-a969-0e729fffc845">

We can mark some sections with `caution` or `danger` over time.

For example, this block should be marked with `:::caution`:
https://pulsar-site-pip-249.vercel.app/docs/next/administration-zk-bk/#configure-bookkeeper

---

<img width="853" alt="Screenshot 2023-05-19 at 10 05 29 AM" src="https://github.com/apache/pulsar-site/assets/9302460/2c08fe4b-87b2-431a-b71c-7c3f1ad6d52b">

---

<img width="843" alt="Screenshot 2023-05-19 at 10 08 55 AM" src="https://github.com/apache/pulsar-site/assets/9302460/002c3ebf-21b4-4e76-9bd0-66dcffd44284">

---

@tisonkun I splitted it into two commits. Just not sure about versioned_docs. Feel free to revert it if needed:
- [Replace :::note to :::info everywhere except of versioned_docs](https://github.com/apache/pulsar-site/pull/560/commits/b190bf687e17526cd69640212fce7484d90b83f8)
- [Replace :::note to :::info in versioned_docs](https://github.com/apache/pulsar-site/pull/560/commits/43b4d8afcc4f9ee49022ccd1790fd90aa7cd2e41)

emidio-cardeira commented:
@visortelle 

**> Logo. Why change this from blue to black?**

We increased a lot the number of colors on the website. So we're trying to make Pulsar visual language more friendlier and approachable.
And because we have all of those colors we think the logo should be more neutral and that way to increase logo visibility.

**Alert banner**
I'm ok in using a light grey like this one, if we need to keep the option.
Design spec here https://www.figma.com/file/0jERoA2DTlfQoj1uH09FxQ/Pulsar-website?type=design&node-id=1165-1046&t=thPf8NEKpouB85vO-4

btw can we remove the shadow please :)



**Body**
This font is a bit smaller that is true
Let's increase the Body P-M style to 
size 18pt
letter spacing -6%
line height 28 

WDYT?

**Titles**
Let's create a new H with 
<img width="949" alt="Screenshot 2023-05-19 at 17 51 51" src="https://github.com/apache/pulsar-site/assets/113345618/7b6d26c2-56b9-457f-a954-ef640c682b9e">

visortelle commented:
@emidio-cardeira 
✅ alert shadow is removed
✅ text font size is increased to 18px
✅ h2 changed to semibold on the docs page

visortelle commented:
@asafm @tisonkun @emidio-cardeira @Anonymitaet

If anyone has anything to add, please do it. 😄
Otherwise, let's merge it. 🚀

asafm commented:
I'm back from vacation. I'll take a look.
Did you end up posting the announcement in the DEV mailing list @visortelle ?

asafm commented:
Looks good to me.
We can plan the next steps after we merge this one.

visortelle commented:
> Did you end up posting the announcement in the DEV mailing list @visortelle ?

@asafm Yes, here it is:
https://lists.apache.org/thread/sjlkpt1zb29k3vfbqvgv271brkf0tcsr

Anonymitaet commented:
Hi @visortelle thanks for your quick updates!

In technical writing, different admonitions have different meanings, they are not exchangeable. 

For example,

> DANGER—Indicates a hazardous situation that, if not avoided, will result in serious problems
> This text is to be limited to the most extreme situations. Call out the hazard and steps to
> avoid it.
> 
> CAUTION—Indicates a hazardous situation that, if not avoided, could result in minor or moderate
> problems. Call out the hazard or consequence and steps to avoid it.
> 
> NOTE—Contains additional information to emphasize or supplement important points of the main
> text. Information in a note should not add value to the preceding text; if it does, use an important
> alert instead.
> 
> TIP—Provides helpful hints for completing a task. Do not use a tip to give essential information. A
> tip might provide an alternative method for completing the task that precedes it.

So can we customize the note color to blue (and not update note to other admonitions)?

visortelle commented:
@Anonymitaet 
I think it depends on the style-guide you prefer to use. :)
I changed it to blue as you wish.

<img width="1141" alt="Screenshot 2023-05-22 at 9 45 23 AM" src="https://github.com/apache/pulsar-site/assets/9302460/69936dc5-9e84-492a-a53d-09a4b502b810">

Anonymitaet commented:
@visortelle that's great! 
I've triggered the tests to run, let's wait for the pass.

tisonkun commented:
Merging...

Thank you!

visortelle commented:
@tisonkun @Anonymitaet at https://pulsar.apache.org I still see the old version.
Maybe we should trigger some CI job to deploy the merged version?

tisonkun commented:
I trigger it at https://github.com/apache/pulsar-site/actions/runs/5053822355/jobs/9068057937

Weird the push commit don't trigger it :/

visortelle commented:
@tisonkun the reason was that one of my commits contained the `[ci skip]` instruction. 
The squashed commit message is a concatenation of all commit messages, therefore all CI jobs were skipped.

<img width="388" alt="Screenshot 2023-05-23 at 8 26 52 AM" src="https://github.com/apache/pulsar-site/assets/9302460/b9a4b39b-970e-4668-a6e9-57eec37b2cca">

tisonkun commented:
I don't cleaning the commit message in this case :D

Anonymitaet commented:
Hi all, 

More developers are checking the new website today, and they think it's awesome!

At the same time, I get feedback from @mattisonchao and @codelipenghui, shown in the image below

<img width="1371" alt="image" src="https://github.com/apache/pulsar-site/assets/50226895/8cab7265-bc70-4d38-aeac-c18faa58976c">

visortelle commented:
Hi @Anonymitaet,

> More developers are checking the new website today, and they think it's awesome!

Glad to hear that. :)

Regarding font size, let's ask @emidio-cardeira.

Screenshots to simplify the comparison:

**2rem (current)**

<img width="1440" alt="Screenshot 2023-05-24 at 8 09 52 AM" src="https://github.com/apache/pulsar-site/assets/9302460/744f115c-c566-40ec-bfc5-dbb74f905787">

**1.5rem**

<img width="1440" alt="Screenshot 2023-05-24 at 8 10 03 AM" src="https://github.com/apache/pulsar-site/assets/9302460/9c5b9596-f761-49ac-9cbd-f0eda5574401">

**1.5rem + less width**

<img width="1440" alt="Screenshot 2023-05-24 at 8 13 13 AM" src="https://github.com/apache/pulsar-site/assets/9302460/74460a24-a26f-4622-a4d5-8ccb26a7201a">

emidio-cardeira commented:
@visortelle @Anonymitaet 
Maybe let's wait for more feedback before making any changes.
let the dust settle a bit and then evaluate
